Code/Source/Frame/ROS2Transform.cpp

```diff
@@ -0,0 +1,88 @@
+/*
+* Copyright (c) Contributors to the Open 3D Engine Project.
+* For complete copyright and license terms please see the LICENSE at the root of this distribution.
+*
+* SPDX-License-Identifier: Apache-2.0 OR MIT
+*
+*/
+
+#include "Frame/ROS2Transform.h"
+#include "Utilities/ROS2Conversions.h"
+#include "ROS2/ROS2Bus.h"
+
+namespace ROS2
+{
+    namespace Internal
+    {
+        class StaticTransformPublisher : public TransformPublisher
+        {
+        public:
+            void Publish(const geometry_msgs::msg::TransformStamped& transformMessage) override
+            {
+                if (m_published)
+                    return; // Consider adding a warning here as well
+
+                return ROS2Interface::Get()->BroadcastStaticTransform(transformMessage);
+                m_published = true;
+            }
+
+        private:
+            bool m_published = false;
+        };
+
+        class DynamicTransformPublisher : public TransformPublisher
+        {
+        public:
+            DynamicTransformPublisher()
+            {
+                auto ros2Node = ROS2Interface::Get()->GetNode();
+
+                // TODO - determine whether default QoS is good and what is good practice for queue size. Compare to tf2 broadcaster
+                const size_t keepLastQueueSize = 10;
+                m_dynamicTFPublisher = ros2Node->create_publisher<geometry_msgs::msg::TransformStamped>("/tf", keepLastQueueSize);
+            }
+
+            void Publish(const geometry_msgs::msg::TransformStamped& transformMessage) override
+            {
+                m_dynamicTFPublisher->publish(transformMessage);
+            }
+
+        private:
+            rclcpp::Publisher<geometry_msgs::msg::TransformStamped>::SharedPtr m_dynamicTFPublisher;
+        };
+
+        AZStd::unique_ptr<TransformPublisher> TransformPublisher::CreateTransformPublisher(bool isDynamic)
+        {
+            if (isDynamic)
+            {
+                return std::make_unique<DynamicTransformPublisher>();
+            }
+            else
+            {
+                return std::make_unique<StaticTransformPublisher>();
```

Comment:
AZStd::make_unique should be used here as there is a [bit of logic](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/std/smart_ptr/unique_ptr.h#L41-L61) to determine whether `aznew` should be used to use the AZ Allocators base on the AZ_CLASS_ALLOCATOR macro